src/types/real.rs

```diff
@@ -0,0 +1,86 @@
+/// Represents a real type in Rust that can be decoded or encoded into any
+/// ASN.1 codec.
+#[cfg(any(feature = "f32", feature = "f64"))]
+pub trait RealType:
+    Sized
+    + core::fmt::Debug
+    + core::fmt::Display
+{
+    /// The byte level width of the floating point type.
+    const BYTE_WIDTH: usize;
+
+    /// Returns the IEEE 754 encoded bytes of the real type, byte count defined in `usize`
+    fn to_ieee754_bytes(&self) -> (impl AsRef<[u8]>, usize);
+
+    /// Attempts to decode the IEEE 754 encoded bytes into `Self`
+    fn try_from_ieee754_bytes(bytes: &[u8]) -> Result<Self, TryFromRealError>;
+
+    /// Converts the real type into a `f64`.
+    fn to_f64(&self) -> f64;
+
+    /// Converts a `f64` into the real type.
+    fn from_f64(value: f64) -> Self;
```

Comment:
These should be `TryInto`/`TryFrom` equivalents meaning that it should allow for the possibility that it fails to convert.